site/gatsby-site/playwright/e2e/unit/gtag.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { expect } from '@playwright/test';
+import config from '../../../config';
+import { test } from '../../utils';
+
+const productionUrl = config.gatsby.siteUrl;
+const googleTrackingId = config.gatsby.gaTrackingId;
+
+test.describe('Google Analytics Tracking', () => {
+  test('Should track gtag only on production', async ({ page }) => {
+    await page.goto('/');
+
+    const currentUrl = page.url();
+
+    const isProduction = currentUrl.startsWith(productionUrl);
+
+    const hasGoogleTracking = await page.locator(`script[src*="https://www.googletagmanager.com/gtag/js?id=${googleTrackingId}"]`).count();
+
+    if (isProduction) {
```

Comment:
I understand the logic here, but I'm trying to see if we can avoid different test flows in different environments.
For example, this `if` condition will evaluate to true in the `Deploy to production` workflow, but it will evaluate to false in all other environments.